lib/msgServer/index.js

```diff
@@ -480,9 +480,13 @@ function setupDiscovery() {
 
 
 	service.on('error', function (error, service) {
-		if (!service) {
+		if (!service && error.name !== 'CONNECTION_LOSS') {
```

Comment:
If there is no service, but the error is CONNECTION_LOSS, we will move on beyond this if-statement and have an uncaught exception on line 493. I expect you'll wanna split this up into 2 cases, one for no service and one for non-CONNECTION_LOSS errors.